clients/client/src/download.js

```diff
@@ -42,10 +42,46 @@ const download = async ({ name, object, streamFactory, retries, delayFactor, ran
   const download = await object.startDownload(name, { acceptDownloadMethods });
 
   if (download.method === 'simple') {
-    return await simple({ download, streamFactory, retryCfg });
+    return await simple({ url: download.url, streamFactory, retryCfg });
   } else {
     throw new Error("Could not negotiate a download method");
   }
 };
 
-module.exports = { download };
+const downloadArtifact = async ({
+  taskId, runId, name, queue, streamFactory, retries, delayFactor, randomizationFactor, maxDelay,
+}) => {
+  const retryCfg = makeRetryCfg({ retries, delayFactor, randomizationFactor, maxDelay });
+
+  let artifact = await (runId === undefined ? queue.latestArtifact(taskId, name) : queue.artifact(taskId, runId, name));
+
+  switch (artifact.storageType) {
+    case "reference":
+    case "s3": {
+      // downloading these artifact types is identical to a simple download
+      return await simple({ url: artifact.url, streamFactory, retryCfg });
+    }
+
+    case "object": {
+      // `taskcluster.Object` is created at runtime, so we cannot require this
```

Comment:
Ah yeah, this is what was making the `import` stuff hard wasn't it?

Reply:
It was one of many things, yeah